### PR TITLE
Split Dynamics.re

### PR DESCRIPTION
**Pull request by annie-anna**
Split Dynamics.re into Expansion.re and Evaluation.re

### CODE REVIEW

File: src/hazelcore/dynamics/Evaluation.re [hunk 1]
```diff
@@ -0,0 +1,370 @@
+open Sexplib.Std /* hole instance numbers are all 0 after expansion and during evaluation -- * renumbering is done on the final result (see below) */;
+
+module Evaluator = {
```
Comment (cyrus-) [inline src/hazelcore/dynamics/Evaluation.re]: we don't need a submodule here, just call this file Evaluator.re

**Review by cyrus- — CHANGES_REQUESTED** [review]

**Comment by dm0n3y** [comment]
One is called `Expansion`, the other is called `Evaluator`. Should we standardize on a noun form, i.e., rename one to `Expander` or other to `Evaluation`? I think it would also make sense to split `Statics` into `Typechecking`/`Typechecker` and `HoleFixing`/`HoleFixer` (not a suggestion to add to this PR, just considering guiding theme)

**Comment by cyrus-** [comment]
let's call it "Expander" and "Evaluator".

regarding hole fixing being its own module, I think it makes more sense as part of Statics -- it behaves the same as the type checker, except it outputs a hole fixed version of the program (in other words it marks the errors, the way a "production" type checker would). if you want to propose it still, go ahead and make a separate issue and we can discuss there.

**Comment by cyrus-** [comment]
actually, while we're at it, let's rename "Expander" to "Elaborator" and "expand" to "elab" in prep for eventually merging in livelits. also its called elaboration in the POPL 19 paper anyway.

**Comment by annie-anna** [comment]
Do I need to rename ``ExpandingKeyword.re`` to ``ElaboratingKeyword.re`` and ``expand`` to ``elab`` in ``PaletteDefinition.re``, too?

**Comment by dm0n3y** [comment]
No, those should stay the same

**Comment by annie-anna** [comment]
Should ``ExpandResult``, ``DoesNotExpand``, ``Expands``, ``UHTyp.expand`` still be the same?

**Comment by dm0n3y** [comment]
`UHTyp.expand` should remain the same, the others should be renamed to `ElaborationResult`, `DoesNotElaborate`, `Elaborates`

**Review by cyrus- — APPROVED** [review]

**Pull request merged**